system/jlib/jsocket.cpp

```diff
@@ -445,22 +449,120 @@ extern jlib_decl bool queryKeepAlive(int &time, int &intvl, int &probes)
                     keepAliveTime = keepalive->getPropInt("@time", keepAliveTime);
                     keepAliveInterval = keepalive->getPropInt("@interval", keepAliveInterval);
                     keepAliveProbes = keepalive->getPropInt("@probes", keepAliveProbes);
-                    state = ENABLED;
+                    hasKeepAlive = true;
                 }
+                disableDNSTimeout = expert->getPropBool("@disableDNSTimeout", false);
+                maxDNSThreads = expert->getPropInt("@maxDNSThreads", maxDNSThreads);
+                // could also consider maxDNSThreads==0 as a way to disable DNS timeout ...
+                if (maxDNSThreads < 1)
+                    maxDNSThreads = 1;
             }
-            doKeepAlive = state;
+            expertTCPSettings = INITED;
         }
     }
+}
 
-    if (state == ENABLED)
+extern jlib_decl bool queryKeepAlive(int &time, int &intvl, int &probes)
+{
+    queryTCPSettings();
+    if (hasKeepAlive)
     {
         time = keepAliveTime;
         intvl = keepAliveInterval;
         probes = keepAliveProbes;
+    }
+    return hasKeepAlive;
+}
+
+static bool getAddressInfo(const char *name, unsigned *netaddr, bool okToLogErr);
+
+static CriticalSection queryDNSCS;
+
+class CAddrInfoThreadArgs : public CInterface, implements IInterface
+{
+public:
+    IMPLEMENT_IINTERFACE;
+    StringAttr name;
+    unsigned netaddr[4] = { 0, 0, 0, 0 };
+    bool result = false;
```

Comment:
technically should this be std::atomic since it is set within the running thread, but checked by the caller.  Since called after join it will not cause any problems.

Reply:
changed.